packages/example-log-extension/src/providers/log-action.provider.ts

```diff
@@ -60,33 +62,37 @@ export class LogActionProvider implements Provider<LogFn> {
       level !== LOG_LEVEL.OFF
     ) {
       if (!args) args = [];
-      let log = `${req.url} :: ${controllerClass.name}.`;
-      log += `${methodName}(${args.join(', ')}) => `;
+      let msg = `${req.url} :: ${controllerClass.name}.`;
+      msg += `${methodName}(${args.join(', ')}) => `;
 
-      if (typeof result === 'object') log += JSON.stringify(result);
-      else log += result;
+      if (typeof result === 'object') msg += JSON.stringify(result);
+      else msg += result;
 
       if (start) {
         const timeDiff: HighResTime = this.timer(start);
         const time: number =
           timeDiff[0] * 1000 + Math.round(timeDiff[1] * 1e-4) / 100;
-        log = `${time}ms: ${log}`;
+        msg = `${time}ms: ${msg}`;
       }
 
       switch (level) {
         case LOG_LEVEL.DEBUG:
-          console.log(chalk.white(`DEBUG: ${log}`));
+          this.log(chalk.white(`DEBUG: ${msg}`));
           break;
         case LOG_LEVEL.INFO:
-          console.log(chalk.green(`INFO: ${log}`));
+          this.log(chalk.green(`INFO: ${msg}`));
           break;
         case LOG_LEVEL.WARN:
-          console.log(chalk.yellow(`WARN: ${log}`));
+          this.log(chalk.yellow(`WARN: ${msg}`));
           break;
         case LOG_LEVEL.ERROR:
-          console.log(chalk.red(`ERROR: ${log}`));
+          this.log(chalk.red(`ERROR: ${msg}`));
           break;
       }
     }
   }
+
+  log(msg: string) {
+    debug(msg);
```

Comment:
I think this makes our example-log-extensions sort of useless. The point of the extension is to let applications produce logs that are useful in production. Now we are logging via `debug` which is almost always turned off in production environments :(

IMO, the log component should:
 - print to console by default
 - allow tests to choose a different target (e.g. `debug`).

Since the problem of cluttered test output is not specific to us only, but will apply to anybody building an application with a logger component, I think it's even more important for our example log component to show an elegant solution for suppressing console logs in tests.

Reply:
I'll take a slab.